File: python/matrix/matrix.py
class Matrix:

    def __init__(self, matrix_string):
        self.rows = [[int(x) for x in row.split()] for row in matrix_string.split("\n")]
        self.columns = [list(l) for l in (zip(*self.rows))]

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return self.columns[index-1]

File: ruby/matrix/matrix.rb
class Matrix
  def initialize(str)
    @str = str
  end

  def rows
    @str.split("\n").map do |arr|
      arr.split.map(&:to_i)
    end
  end

  def columns
    @rows.first.zip(*rows[1..])
  end
end
